test/Dialect/Tpp/tpp-bufferization.mlir

```diff
@@ -422,3 +381,127 @@ func.func @brgemm_in_loops(%arg0: tensor<4x16x32x32xf32>, %arg1: tensor<8x16x32x
+    %11 = tpp.relu (%10 : tensor<32x32xf32>) -> (tensor<32x32xf32>)
+    scf.forall.in_parallel {
+      tensor.parallel_insert_slice %11 into %arg3[%arg1, %arg2] [32, 32] [1, 1] : tensor<32x32xf32> into tensor<256x1024xf32>
+    }
+  }
+  %3 = tensor.empty() : tensor<256x1024xf32>
+  %4 = tpp.zero (%3 : tensor<256x1024xf32>) -> (tensor<256x1024xf32>)
+  %5 = scf.forall (%arg1, %arg2) = (%c0, %c0) to (%c256, %c1024) step (%c32, %c32) shared_outs(%arg3 = %4) -> (tensor<256x1024xf32>) {
+    %extracted_slice = tensor.extract_slice %2[%arg1, 0] [32, 1024] [1, 1] : tensor<256x1024xf32> to tensor<32x1024xf32>
+    %extracted_slice_3 = tensor.extract_slice %arg3[%arg1, %arg2] [32, 32] [1, 1] : tensor<256x1024xf32> to tensor<32x32xf32>
+    %9 = tpp.matmul (%extracted_slice : tensor<32x1024xf32>, %cst_1 : tensor<1024x32xf32>, %extracted_slice_3 : tensor<32x32xf32>) -> (tensor<32x32xf32>)
+    %extracted_slice_4 = tensor.extract_slice %arg3[%arg1, %arg2] [32, 32] [1, 1] : tensor<256x1024xf32> to tensor<32x32xf32>
+    %10 = tpp.add (%9 : tensor<32x32xf32>, %cst_0 : tensor<32x32xf32>) -> (tensor<32x32xf32>)
+    %extracted_slice_5 = tensor.extract_slice %arg3[%arg1, %arg2] [32, 32] [1, 1] : tensor<256x1024xf32> to tensor<32x32xf32>
+    %11 = tpp.relu (%10 : tensor<32x32xf32>) -> (tensor<32x32xf32>)
+    scf.forall.in_parallel {
+      tensor.parallel_insert_slice %11 into %arg3[%arg1, %arg2] [32, 32] [1, 1] : tensor<32x32xf32> into tensor<256x1024xf32>
+    }
+  }
+  %6 = tensor.empty() : tensor<256x1024xf32>
+  %7 = tpp.zero (%6 : tensor<256x1024xf32>) -> (tensor<256x1024xf32>)
+  %8 = scf.forall (%arg1, %arg2) = (%c0, %c0) to (%c256, %c1024) step (%c32, %c32) shared_outs(%arg3 = %7) -> (tensor<256x1024xf32>) {
+    %extracted_slice = tensor.extract_slice %5[%arg1, 0] [32, 1024] [1, 1] : tensor<256x1024xf32> to tensor<32x1024xf32>
+    %extracted_slice_3 = tensor.extract_slice %arg3[%arg1, %arg2] [32, 32] [1, 1] : tensor<256x1024xf32> to tensor<32x32xf32>
+    %9 = tpp.matmul (%extracted_slice : tensor<32x1024xf32>, %cst_1 : tensor<1024x32xf32>, %extracted_slice_3 : tensor<32x32xf32>) -> (tensor<32x32xf32>)
+    %extracted_slice_4 = tensor.extract_slice %arg3[%arg1, %arg2] [32, 32] [1, 1] : tensor<256x1024xf32> to tensor<32x32xf32>
+    %10 = tpp.add (%9 : tensor<32x32xf32>, %cst_2 : tensor<32x32xf32>) -> (tensor<32x32xf32>)
+    %extracted_slice_5 = tensor.extract_slice %arg3[%arg1, %arg2] [32, 32] [1, 1] : tensor<256x1024xf32> to tensor<32x32xf32>
+    %11 = tpp.relu (%10 : tensor<32x32xf32>) -> (tensor<32x32xf32>)
+    scf.forall.in_parallel {
+      tensor.parallel_insert_slice %11 into %arg3[%arg1, %arg2] [32, 32] [1, 1] : tensor<32x32xf32> into tensor<256x1024xf32>
+    }
+  }
+  return %8 : tensor<256x1024xf32>
+}
+
+// -----
+
+// CHECK-LABEL: alloc_must_escape
+func.func @alloc_must_escape(%arg0: tensor<256x1024xf32>) -> tensor<256x1024xf32> {
```

Comment:
Could you add a small comment what it tests? I'll forget tomorrow and will be confused when it breaks 😅

Reply:
Maybe a word why it would inject dealloc in the first place.

Reply:
This was due to a bug in `getAliasingOpResultsBinaryImpl` which was not setting proper alias information, thus bufferization was inserting the dealloc. I made explicit in the IR what we expect from bufferization.